### PR TITLE
Auto generate API reference docs

### LINKED ISSUE #178: Generate documentation for Astarte CRDs
CRDs are documented within the code. Make them easily available for users to consult.

### PR DESCRIPTION
**Pull request by Annopaolo**
Add API reference docs generation from sources using [crd-ref-docs](https://github.com/elastic/crd-ref-docs). 
Also, add a workflow to automatically deploy newly generated docs to [astarte-platform/docs](https://github.com/astarte-platform/docs) using [Hugo static pages generator](https://gohugo.io/).
Fix #178 .
Istantanea_2022-02-23_12-42-59

### CODE REVIEW

**Comment by Annopaolo** [comment]
Blocked until  #235 is merged.

**Review by matt-mazzucato — APPROVED** [review]

File: docs/autogen/config.yaml
Comment (bettio) [inline docs/autogen/config.yaml:3]: do they make sense for us? I don't think so

File: docs/content/index.md
Comment (bettio) [inline docs/content/index.md:1]: are we versioning generated doc?

**Review by bettio — CHANGES_REQUESTED** [review]

File: docs/content/index.md
Comment (Annopaolo) [reply docs/content/index.md]: We now generate that file in the workflow

**Review by bettio — APPROVED** [review]

**Pull request merged**
